INSS.EIIR.DataSync.Application/UseCase/SyncData/SyncData.cs

```diff
@@ -13,27 +14,58 @@ public class SyncData : IResponseUseCase<SyncDataResponse>
         private readonly TransformService _transformService;
         private readonly ValidationService _validation;
         private readonly ILogger<SyncData> _logger;
+        private bool _swapIndexAndZipXml = true;
+        private readonly IExtractRepository _eiirRepository;
 
-        public SyncData(SyncDataOptions options, ILogger<SyncData> logger)
+        public SyncData(SyncDataOptions options, IExtractRepository extractRepository, ILogger<SyncData> logger)
         {
             _options = options;
             _logger = logger;
             _transformService = new TransformService(_options.TransformRules);
             _validation = new ValidationService();
+            _eiirRepository = extractRepository;
         }
 
         public async Task<SyncDataResponse> Handle()
         {
+            var resp = new SyncDataResponse() { ErrorCount = 0, ErrorMessage = string.Empty };
+
             int numErrors = 0;
 
+            await _options.FailureSink.Start();
+
```

Comment:
can we move the pre-conditions check to separate method

Reply:
Have pulled preconditions into separate private methods... looks better.